### PR TITLE
Implement cmd as List

### LINKED ISSUE #91: Gather commands to a list
#### Summary:

Calling subprocess needs a list of commands. 
Currently `_run_cmdstan` functions uses `.split` method to create a list from the cmd string.
If one of the items contains whitespace (a path for example) then the subprocess will fail.

https://github.com/stan-dev/cmdstanpy/blob/9dd4b45788e0e59ad2230ec60b854dbf592bf6c4/cmdstanpy/model.py#L566

### LINKED ISSUE #167: Spaces in directory path cause errors in sampling
#### Summary:
I was at the tutorial session at PyData NYC on Wednesday afternoon. I'm getting an error running some of the tutorial code on my Mac (but can run it fine on Linux).

#### Description:
Here is the python code, stan code, and data that are causing the error:

```python
import os
from cmdstanpy import CmdStanModel
import pandas as pd

season_1975 = pd.read_csv('efron-morris-75-data.csv')
data_dict = {'N': season_1975.shape[0], 'y' : season_1975['Hits'].tolist(), 'K' : season_1975['At-Bats'].tolist()}

model_complete_pool = CmdStanModel(stan_file='simple_pool.stan')
model_complete_pool.compile()
complete_pool_fit = model_complete_pool.sample(data=data_dict)
complete_pool_fit.summary().round(decimals=2)
```

```
FirstName,LastName,At-Bats,Hits,BattingAverage,RemainingAt-Bats,RemainingAverage,SeasonAt-Bats,SeasonHits,SeasonAverage
Roberto,Clemente,45,18,0.4,367,0.346,412,145,0.352
Frank,Robinson,45,17,0.378,426,0.2981,471,144,0.306
Frank,Howard,45,16,0.356,521,0.2764,566,160,0.283
Jay,Johnstone,45,15,0.333,275,0.2218,320,76,0.238
Ken,Berry,45,14,0.311,418,0.2727,463,128,0.276
Jim,Spencer,45,14,0.311,466,0.2704,511,140,0.274
Don,Kessinger,45,13,0.289,586,0.2645,631,168,0.266
Luis,Alvarado,45,12,0.267,138,0.2101,183,41,0.224
Ron,Santo,45,11,0.244,510,0.2686,555,148,0.267
Ron,Swaboda,45,11,0.244,200,0.23,245,57,0.233
Rico,Petrocelli,45,10,0.222,538,0.2639,583,152,0.261
Ellie,Rodriguez,45,10,0.222,186,0.2258,231,52,0.225
George,Scott,45,10,0.222,435,0.3034,480,142,0.296
Del,Unser,45,10,0.222,277,0.2635,322,83,0.258
Billy,Williams,45,10,0.222,591,0.3299,636,205,0.251
Bert,Campaneris,45,9,0.2,558,0.2849,603,168,0.279
Thurman,Munson,45,8,0.178,408,0.3162,453,137,0.302
Max,Alvis,45,7,0.156,70,0.2,115,21,0.183
```

```stan
data {
  int<lower=0> N;           // items
  int<lower=0> K[N];        // initial trials
  int<lower=0> y[N];        // initial successes
}
parameters {
  real<lower=0, upper=1> phi;  // chance of success (pooled)
}
model {
  y ~ binomial(K, phi);  // likelihood
}
```

The error I am getting is the following:
```
INFO:cmdstanpy:stan to c++ (/var/folders/yl/v6d742w56rjdcr406hdxkfpr0000gn/T/tmpuy6wrr9z/tmpwf4d7zmm.hpp)
INFO:cmdstanpy:compiling c++
INFO:cmdstanpy:compiled model file: simple_pool
INFO:cmdstanpy:start chain 1
INFO:cmdstanpy:start chain 2
INFO:cmdstanpy:start chain 3
INFO:cmdstanpy:start chain 4
Traceback (most recent call last):
  File "test.py", line 10, in <module>
    complete_pool_fit = model_complete_pool.sample(data=data_dict)
  File "/Users/jwbarnet/Documents/PyData NYC 2019/bayesian_inference/venv/lib/python3.7/site-packages/cmdstanpy/model.py", line 615, in sample
    raise RuntimeError(msg)
RuntimeError: Error during sampling, chain 0 returned error code -1, chain 1 returned error code -1, chain 2 returned error code -1, chain 3 returned error code -1
deleting tmpfiles dir: /var/folders/yl/v6d742w56rjdcr406hdxkfpr0000gn/T/tmpjwvcen19
done
```

#### Additional Information:
This error only occurs on MacOS. It does not seem to be a problem on my Linux machine. I've tried reinstalling both cmdstan and cmdstanpy.

#### Current Version:
cmdstanpy: 0.6.0
cmdstan: 2.21.0

### PR DESCRIPTION
**Pull request by wesbarnett**
#### Submission Checklist

- [x] Run unit tests
- [x] Declare copyright holder and open-source license: see below

#### Summary

Changes command into a List. Previously it was a String and `.split()` was used. In addition an absolute path is used for the target stanfile in the scenario where it is copied from a tmpdir.

Fixes #167 and fixes #91.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): American Express

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)

### CODE REVIEW

**Comment by wesbarnett** [comment]
Sure, I can add that. Would you prefer that in the same PR or a separate one?

**Comment by mitzimorris** [comment]
same PR, please

File: test/test_sample.py [hunk 1]
```diff
@@ -245,6 +245,80 @@ def test_fixed_param_good(self):
         self.assertEqual(datagen_fit.metric_type, None)
         self.assertEqual(datagen_fit.stepsize, None)
 
+    def test_bernoulli_file_with_space(self):
+        stan = os.path.join(datafiles_path, 'bernoulli with space in name.stan')
```
Comment (ahartikainen) [inline test/test_sample.py]: Is there a difference with filename with space and filepath with space?

Comment (wesbarnett) [reply test/test_sample.py]: Interestingly, there might be. I get different error codes without the patch for the case of a file with spaces vs a directory with spaces. In both cases this PR fixes it. So, I can add another test for a path with spaces.

**Comment by wesbarnett** [comment]
I'm essentially copying the good test case and modifying the filename/path for the unittest. Maybe there is a better approach with that so that the same test could be re-used with just changing the stan file name/path?

**Comment by mitzimorris** [comment]
> Maybe there is a better approach with that so that the same test could be re-used with just changing the stan file name/path?

http://doc.pytest.org/en/latest/reference.html#pytest-mark-parametrize-ref

parameterize a function - call it several times with path/filenames?

**Comment by wesbarnett** [comment]
> > Maybe there is a better approach with that so that the same test could be re-used with just changing the stan file name/path?
> 
> http://doc.pytest.org/en/latest/reference.html#pytest-mark-parametrize-ref
> 
> parameterize a function - call it several times with path/filenames?

Perfect. Will take a look.

**Comment by wesbarnett** [comment]
It looks like pytest's parametrize doesn't work with unittest.TestCase: https://docs.pytest.org/en/latest/unittest.html

**Comment by wesbarnett** [comment]
Since pytest's parametrize doesn't work with unittest.TestCase, I made the name of the stanfile an argument to the test and created two new tests with two different filenames. The relevant changes are [here](https://github.com/stan-dev/cmdstanpy/pull/168/files#diff-cfd6161e92523bda38afdf3dfb8ab2c3).

**Pull request merged**